doc/nrf/libraries/modem/at_parser.rst

```diff
@@ -0,0 +1,103 @@
+.. _at_parser_readme:
+
+AT parser
+#########
+
+.. contents::
+   :local:
+   :depth: 2
+
+Overview
+========
+
+The AT parser is a library that parses AT command responses, notifications, or events.
+It exports an API that lets the user iterate through an AT command string and retrieve any of its elements based on their indices, efficiently and without allocations.
+
+Initialization
+==============
+
+To begin using the AT parser, you need to first initialize it. This is done by invoking the :c:func:`at_parser_init` function, where you provide a pointer to the AT parser and the AT command string that needs to be parsed.
+
+The following code snippet shows how to initialize an AT parser for an AT command response:
+
+.. code-block:: c
+
+   int err;
+   struct at_parser parser = { 0 };
```

Comment:
One comment here, `at_parser_init()` zeros and then initializes the parser instance. Why is then the parser always zero-initialized upon declaration? That way it is actually zeroed twice, which is not necessary.

Reply:
Updated.